pkg/apis/cert/v1alpha1/types.go

```diff
@@ -93,6 +96,43 @@ type IssuerRef struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
+// PrivateKeyAlgorithm is the type for the algorithm
+// +kubebuilder:validation:Enum=RSA;ECDSA
+type PrivateKeyAlgorithm string
+
+const (
+	// RSAKeyAlgorithm is the value to use the RSA algorithm for the private key.
+	RSAKeyAlgorithm PrivateKeyAlgorithm = "RSA"
+
+	// ECDSAKeyAlgorithm is the value to use the ECDSA algorithm for the private key.
+	ECDSAKeyAlgorithm PrivateKeyAlgorithm = "ECDSA"
+)
+
+// CertificatePrivateKey contains configuration options for private keys
+// used by the Certificate controller.
+// These include the key algorithm and size.
+type CertificatePrivateKey struct {
+	// Algorithm is the private key algorithm of the corresponding private key
+	// for this certificate.
+	//
+	// If provided, allowed values are either `RSA` or `ECDSA`.
+	// If `algorithm` is specified and `size` is not provided,
+	// key size of 2048 will be used for `RSA` key algorithm and
+	// key size of 256 will be used for `ECDSA` key algorithm.
+	// +optional
```

Comment:
Can we work with `// +kubebuilder:validation:Enum=RSA;ECDSA` here, maybe also for the `size`?

Reply:
Interestingly, you can really define an enumeration for numbers.🤩